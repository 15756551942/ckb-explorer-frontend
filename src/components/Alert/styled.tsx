import styled from 'styled-components'

export const AlertPanel = styled.div`
  position: sticky;
  position: -webkit-sticky;
  top: 0;
  z-index: 9000;

  > div {
    width: 100%;
    height: 48px;
    background: #fa8f00;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0 120px;
    color: white;
    font-size: 14px;
    font-weight: 450;

    @media (max-width: 1440px) {
      padding: 0 100px;
    }

    @media (max-width: 1200px) {
      padding: 0 45px;
    }

    @media (max-width: 750px) {
      padding: 0 18px;
    }
  }

  .alert__dismiss {
    width: 100px;
    height: 30px;
    line-height: 30px;
    border-radius: 2px;
    border: solid 1px #ffffff;
    text-align: center;
  }
`
