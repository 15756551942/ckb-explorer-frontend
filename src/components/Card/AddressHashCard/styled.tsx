import styled from 'styled-components'

export default styled.div`
  width: 100%;
  border-radius: 6px;
  box-shadow: 2px 2px 6px 0 #dfdfdf;
  background-color: #ffffff;
  height: 80px;
  padding: 0px 0px 24px 0px;
  display: flex;
  flex-direction: row;
  align-items: flex-end;

  @media (max-width: 700px) {
    height: 50px;
    padding: 0px 0px 17px 0px;
    border-radius: 3px;
    box-shadow: 1px 1px 3px 0 #dfdfdf;
  }

  .address_hash__title {
    margin-left: 40px;
    font-family: Montserrat;
    font-size: 30px;
    font-weight: 500;
    color: #000000;
    height: 36px;

    @media (max-width: 700px) {
      font-size: 15px;
      margin-left: 20px;
      height: 16px;
    }
  }

  #address_hash__hash {
    margin-left: 20px;
    font-family: Montserrat;
    font-size: 20px;
    color: #000000;
    height: 24px;

    @media (max-width: 700px) {
      font-size: 14px;
      margin-left: 10px;
      height: 16px;
      font-weight: 500;
    }
  }

  .address_hash__copy_iocn {
    width: 21px;
    height: 24px;
    cursor: pointer;
    margin-left: 20px;

    > img {
      width: 100%;
      height: 100%;
    }

    @media (max-width: 700px) {
      margin-left: 10px;
      width: 16px;
      height: 18px;
    }
  }
`
