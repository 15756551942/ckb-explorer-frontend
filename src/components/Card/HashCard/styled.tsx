import styled from 'styled-components'

export const HashCardPanel = styled.div`
  width: 100%;
  height: auto;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  box-shadow: 2px 2px 6px 0 #dfdfdf;
  border-radius: 6px;
  background-color: #ffffff;
  padding: 0 40px;
  @media (max-width: 750px) {
    padding: 0 16px;
    border-radius: 3px;
    box-shadow: 1px 1px 3px 0 #dfdfdf;
  }
  .hash__card__content__panel {
    width: 100%;
    height: 80px;
    display: flex;
    flex-direction: row;
    align-items: center;
    overflow: hidden;
    position: relative;
    @media (max-width: 750px) {
      height: auto;
      flex-direction: ${(props: { isColumn: boolean }) => (props.isColumn ? 'column' : 'row')};
      align-items: ${(props: { isColumn: boolean }) => (props.isColumn ? 'flex-start' : 'center')};
      padding-top: 8px;
      padding-bottom: 8px;
    }
  }
  .hash__icon {
    width: 40px;
    height: 40px;
    margin-right: 8px;
  }
  .hash__title {
    font-size: 26px;
    font-weight: 600;
    color: #000000;
    white-space: nowrap;
    @media (max-width: 750px) {
      font-size: 15px;
    }
  }
  .hash__card__hash__content {
    display: flex;
    flex-direction: row;
    align-items: center;
  }
  #hash__text {
    margin-left: 20px;
    font-size: 18px;
    color: #000000;
    transform: translateY(3px);
    @media (max-width: 750px) {
      font-size: 13px;
      margin-left: ${(props: { isColumn: boolean }) => (props.isColumn ? '0px' : '10px')};
      font-weight: 500;
      transform: translateY(1px);
    }
  }
  .hash__copy_icon {
    cursor: pointer;
    margin-left: 20px;
    transform: translateY(6px);
    @media (max-width: 750px) {
      margin-left: 10px;
      transform: translateY(3px);
    }
    > img {
      width: 21px;
      height: 24px;
      @media (max-width: 750px) {
        width: 16px;
        height: 18px;
        margin-bottom: 3px;
      }
    }
  }
  a {
    margin-left: 10px;
    font-size: 18px;
    margin-top: 8px;
    color: ${props => props.theme.primary};
    @media (max-width: 750px) {
      margin-left: 5px;
      font-size: 12px;
      margin-top: 4px;
    }
  }
  a:hover {
    color: ${props => props.theme.primary};
  }
  #hash__value {
    color: #ffffff;
    position: absolute;
    bottom: -30px;
  }
`

export const LoadingPanel = styled.div`
  width: 100%;
`
