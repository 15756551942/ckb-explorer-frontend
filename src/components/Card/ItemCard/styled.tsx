import styled from 'styled-components'

export const ItemCardPanel = styled.div`
  width: 100%;
  background-color: #ffffff;
  color: #000000;
  font-size: 16px;
  margin-top: 20px;
  border-radius: 0px 0px 6px 6px;
  box-shadow: 2px 2px 6px 0 #dfdfdf;
  padding: 15px 40px;

  @media (max-width: 1000px) {
    font-size: 13px;
  }

  @media (max-width: 750px) {
    font-size: 13px;
    border-radius: 0px 0px 3px 3px;
    box-shadow: 1px 1px 3px 0 #dfdfdf;
    padding: 10px 20px;
  }
`

export const ItemContentPanel = styled.div`
  margin-right: 45px;
  display: flex;
  flex: 1;
  flex-direction: column;

  @media (max-width: 1200px) {
    width: 100%;
    margin-right: 0px;
  }
`

export const ItemDetailPanel = styled.div`
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: space-between;
  position: relative;

  @media (min-width: 750px) {
    height: 20px;
    margin-top: 14px;
  }

  @media (min-width: 1200px) {
    height: 20px;
    margin-top: 14px;
  }

  @media (max-width: 750px) {
    justify-content: normal;
    flex-direction: column;
    align-items: flex-start;

    &:after {
      content: '';
      background: #e2e2e2;
      height: 1px;
      width: 100%;
      margin: 10px 0px 0px 0px;
      display: ${({ hideLine }: { hideLine: boolean }) => (hideLine ? 'none' : 'block')};
      transform: ${() => `scaleY(${Math.ceil((1.0 / window.devicePixelRatio) * 10.0) / 10.0})`};
    }
  }

  .item__detail__title {
    font-size: 16px;
    color: rgba(0, 0, 0, 0.6);
    margin-left: 0px;

    @media (max-width: 750px) {
      width: 100%;
      margin-left: 0px;
    }
  }

  .item__detail__value {
    margin-left: 15px;
    display: flex;
    font-size: 16px;
    color: #000000;

    @media (max-width: 750px) {
      margin-left: 0px;
      word-wrap: break-word;
      word-break: break-all;
      width: 100%;
    }

    a {
      color: ${props => props.theme.primary};
    }

    a:hover {
      color: ${props => props.theme.primary};
    }
  }

  .block_pointer {
    cursor: pointer;
  }
`
