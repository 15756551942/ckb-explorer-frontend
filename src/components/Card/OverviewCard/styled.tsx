import styled from 'styled-components'

export const OverviewCardPanel = styled.div`
  width: 100%;
  border-radius: 0px 0px 6px 6px;
  box-shadow: 2px 2px 6px 0 #dfdfdf;
  background-color: #ffffff;
  margin-top: 5px;
  padding: 10px 40px 30px 40px;
  background-color: #ffffff;

  /* common */
  color: #000000;
  font-size: 16px;

  @media (max-width: 1000px) {
    font-size: 13px;
  }

  @media (max-width: 750px) {
    border-radius: 0px 0px 3px 3px;
    box-shadow: 1px 1px 3px 0 #dfdfdf;
    padding: 5px 20px 15px 20px;

    font-size: 13px;
  }
`

export const OverviewContentPanel = styled.div`
  display: flex;
  flex-direction: row;
  align-items: flex-start;

  @media (max-width: 750px) {
    flex-direction: column;
  }

  > span {
    width: 1px;
    height: ${({ length }: { length: number }) => `${length * 40 - 20}px`};
    background: #e2e2e2;
    margin: 20px 0px 0px 0px;
    transform: ${() => `scaleX(${Math.ceil((1.0 / window.devicePixelRatio) * 10.0) / 10.0})`};

    @media (max-width: 750px) {
      display: none;
    }
  }

  .overview_content__left_items {
    margin-right: 45px;
    display: flex;
    flex: 1;
    flex-direction: column;

    @media (max-width: 750px) {
      width: 100%;
      margin-right: 0px;
    }
  }

  .overview_content__right_items {
    margin-left: 45px;
    display: flex;
    flex: 1;
    flex-direction: column;

    @media (max-width: 750px) {
      width: 100%;
      margin-left: 0px;
    }
  }
`

export const OverviewItemPanel = styled.div`
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: space-between;
  position: relative;

  @media (min-width: 750px) {
    height: 20px;
    margin-top: 20px;
  }

  @media (max-width: 750px) {
    margin-top: 10px;
    justify-content: normal;
    flex-direction: column;
    align-items: flex-start;

    &:after {
      content: '';
      background: #e2e2e2;
      height: 1px;
      width: 100%;
      display: ${({ hiddenLine }: { hiddenLine: boolean }) => (hiddenLine ? 'none' : 'block')};
      margin: 10px 0px 0px 0px;

      transform: ${() => `scaleY(${Math.ceil((1.0 / window.devicePixelRatio) * 10.0) / 10.0})`};
    }
  }

  .overview_item__title {
    font-weight: 500;

    @media (max-width: 750px) {
      width: 100%;
    }
  }

  .overview_item__value {
    margin-left: 15px;
    display: flex;

    @media (max-width: 750px) {
      margin-left: 0px;
      word-wrap: break-word;
      word-break: break-all;
      width: 100%;
    }

    a {
      color: ${props => props.theme.primary};
    }

    a:hover {
      color: ${props => props.theme.primary};
    }
  }

  .block_pointer {
    cursor: pointer;
  }
`
