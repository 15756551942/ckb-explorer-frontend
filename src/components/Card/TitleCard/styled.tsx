import styled from 'styled-components'

export default styled.div`
  width: 100%;
  border-radius: 6px 6px 0px 0px;
  box-shadow: 2px 2px 6px 0 #dfdfdf;
  background-color: #ffffff;
  height: 50px;
  margin-top: 20px;
  padding: 15px 0px;
  color: #000000;
  font-family: Montserrat;
  font-size: 20px;
  font-weight: 500;
  line-height: 1;
  text-align: center;

  @media (max-width: 700px) {
    margin-top: 10px;
    padding: 10px 0px;
    height: 40px;
    font-size: 16px;
    border-radius: 3px 3px 0px 0px;
    box-shadow: 1px 1px 3px 0 #dfdfdf;
  }
`
