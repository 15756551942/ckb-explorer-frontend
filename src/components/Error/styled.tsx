import styled from 'styled-components'

export const ErrorPanel = styled.div`
  margin: 112px 0 203px 0;

  @media (max-width: 750px) {
    margin: 120px 0 130px 0;
  }

  > img {
    width: 1038px;
    height: 480px;
    margin: 0 auto;

    @media (max-width: 750px) {
      width: 282px;
      height: 130px;
    }
  }
`
