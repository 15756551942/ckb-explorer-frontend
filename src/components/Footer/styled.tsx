import styled from 'styled-components'

export const FooterPanel = styled.div`
  font-family: source-code-pro, Menlo, Monaco, Consolas, Courier New, monospace;
  background-color: #000000;
  display: flex;
  flex-direction: column;
  align-items: center;

  .footer__copyright {
    height: 15px;
    font-size: 12px;
    color: #acacac;
    margin-bottom: 28px;
    display: flex;
    justify-content: center;
  }

  a:hover {
    color: ${props => props.theme.primary};
  }
`

export const FooterMenuPanel = styled.div`
  width: 100%;
  overflow: hidden;
  margin: 38px auto 34px auto;
  display: flex;
  justify-content: center;

  .footer__foundation {
    display: flex;
    flex-direction: column;
  }

  .footer__developer {
    display: flex;
    flex-direction: column;
    margin-left: 280px;

    @media (max-width: 1980px) {
      margin-left: 178px;
    }

    @media (max-width: 1440px) {
      margin-left: 138px;
    }

    @media (max-width: 1200px) {
      margin-left: 78px;
    }
  }

  .footer__community {
    display: flex;
    flex-direction: column;
    margin-left: 280px;

    @media (max-width: 1980px) {
      margin-left: 178px;
    }

    @media (max-width: 1440px) {
      margin-left: 138px;
    }

    @media (max-width: 1200px) {
      margin-left: 78px;
    }

    > div {
      display: flex;
      flex-direction: row;
    }
  }

  .footer__title {
    font-family: SourceCodePro;
    font-size: 28px;
    font-weight: bold;
    color: #ffffff;
  }
`

export const FooterItemPanel = styled.a`
  font-size: 18px;
  color: #acacac;
  height: 23px;
  margin: 8px 0;
`

export const FooterImageItemPanel = styled.a`
  width: 60px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 12px;
  color: #acacac;

  &:hover {
    color: ${props => props.theme.primary};
  }

  > img {
    width: 35px;
    height: 35px;
  }

  > span {
    height: 15px;
    font-size: 12px;
    margin-top: 5px;
  }
`
