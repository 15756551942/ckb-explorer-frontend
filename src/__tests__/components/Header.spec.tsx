import React, { ReactElement } from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Header from '../../components/Header'
import { HeaderPanel } from '../../components/Header/styled'
import { BrowserRouter } from 'react-router-dom'

describe('Header Component', () => {
  let component: ReactElement

  beforeAll(() => {
    component = (
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    )
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('Component Render', () => {
    const wrapper = shallow(component)

    expect(wrapper).toBeDefined()
    expect(wrapper.find(Header)).toHaveLength(1)
  })
})
