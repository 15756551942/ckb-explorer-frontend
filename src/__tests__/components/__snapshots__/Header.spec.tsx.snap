// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component shallow renders 1`] = `
<div
  className="sc-AxjAm knJacL"
>
  <a
    className="sc-AxirZ fbJJZa"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="logo"
      src="ckb_logo.png"
    />
  </a>
  <div
    className="sc-AxheI gjDwdY"
  >
    <a
      className="header__menus__item"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="header__menus__item"
      href="/nervosdao"
      onClick={[Function]}
    >
      Nervos DAO
    </a>
    <a
      className="header__menus__item"
      href="/tokens"
      onClick={[Function]}
    >
      Tokens
    </a>
    <a
      className="header__menus__item"
      href="/charts"
      onClick={[Function]}
    >
      Charts
    </a>
  </div>
  <div
    className="sc-AxiKw cjyCOd"
  />
  <div
    className="sc-fzqNJr jOQjll"
    id="header__blockchain__panel"
    onMouseLeave={[Function]}
  >
    <div
      className="sc-fzokOt dREVrx header__blockchain__flag"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseOver={[Function]}
      role="button"
      tabIndex={-1}
    >
      <div
        className="header__blockchain__content_panel"
      >
        <div
          className="header__blockchain__content"
        >
          LINA
        </div>
        <img
          alt="dropdown icon"
          src="white_dropdown.png"
        />
      </div>
      <div
        className="header__blockchain__node__version"
      >
        
      </div>
    </div>
  </div>
  <div
    className="sc-fznKkj iwKOjr"
    id="header__language__panel"
    onMouseLeave={[Function]}
  >
    <div
      className="sc-fzokOt dREVrx header__language__flag"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseOver={[Function]}
      role="button"
      tabIndex={-1}
    >
      <div
        className="header__language__content_panel"
      >
        <div
          className="header__language__content"
        >
          简体中文
        </div>
        <img
          alt="dropdown icon"
          src="white_dropdown.png"
        />
      </div>
    </div>
  </div>
</div>
`;
